protocol/avdl/keybase1/contacts.avdl

```diff
@@ -0,0 +1,44 @@
+@namespace("keybase.1")
+protocol contacts {
+    import idl "common.avdl";
+
+    // Contact syncing / resolving
+    // ---------------------------
+
+    record ContactComponent {
+        // One of:
+        union { null, RawPhoneNumber } phoneNumber;
+        union { null, EmailAddress } email;
+        string label; // "", "Home", "Work"
+    }
+    record Contact {
+        string name; // "Bob Goodman"
+        array<ContactComponent> components;
+    }
+    record ResolvedContact {
+        string name; // "Bob Goodman"
```

Comment:
talked with @cecileboucheron, and we decided that this field should default to the contact name from the contact list, fallback to email if empty and otherwise fall back to phonenumber so that a user can know the mapping between a keybase user and what they called the user in their contact list

Reply:
also, if we fall back to phone or email we should use the `ContactComponent.Label` field in the formatting if `len(Contact.Components) > 0`(zeplin designs show an example)